docs/aws/audit/iamcompliance/rules/role_inline_policies_remediation.mdx

```diff
@@ -1,6 +1,203 @@
+
+def main():
+    roles = list_roles()
+    for role in roles:
+        remove_inline_policies(role['RoleName'])
+
+if __name__ == "__main__":
+    main()
+```
+
+### Summary
+1. **Install Boto3**: Ensure Boto3 is installed in your Python environment.
+2. **Initialize Boto3 Client**: Set up the IAM client using Boto3.
+3. **List All Roles**: Retrieve all IAM roles in your AWS account.
+4. **Check and Remove Inline Policies**: For each role, check for inline policies and remove them if they exist.
+
+This script will help you prevent IAM roles from having inline policies by removing any existing inline policies. You can run this script periodically or integrate it into your CI/CD pipeline to ensure compliance.
+</Accordion>
+
+</AccordionGroup>
+</Tab>
+<Tab title='Cause'>
+### Check Cause
+<AccordionGroup>
+<Accordion title='Using Console' defaultOpen='true'>
+1. Sign in to the AWS Management Console and open the IAM console at https://console.aws.amazon.com/iam/.
+
+2. In the navigation pane, choose "Roles". This will display a list of all the IAM roles that are currently configured in your AWS environment.
+
+3. Click on the name of the role that you want to inspect. This will open the summary page for the selected IAM role.
+
+4. In the "Permissions" tab, look for the "Inline Policies" section. If there are any inline policies attached to the role, they will be listed here. If this section is empty, it means that the role does not have any inline policies, which is the recommended configuration.
+</Accordion>
+
+<Accordion title='Using CLI'>
+1. First, you need to install and configure AWS CLI on your local machine. You can do this by following the instructions provided by AWS. Once you have AWS CLI installed and configured, you can start using it to interact with AWS services.
+
+2. To list all the IAM roles in your AWS account, use the following command:
+
+   ```
+   aws iam list-roles
+   ```
+
+   This command will return a JSON object that contains all the IAM roles in your AWS account.
+
+3. To check if a role has inline policies, you can use the following command:
+
+   ```
+   aws iam list-role-policies --role-name {role-name}
+   ```
+
+   Replace `{role-name}` with the name of the role you want to check. This command will return a list of all inline policies attached to the specified role. If the list is empty, it means that the role does not have any inline policies.
+
+4. Repeat step 3 for each role in your AWS account. If any role has inline policies, it means that there is a misconfiguration.
+</Accordion>
+
+<Accordion title='Using Python'>
+1. First, you need to install the AWS SDK for Python (Boto3) if you haven't done so already. You can install it using pip:
+
+```python
+pip install boto3
+```
+
+2. Import the necessary modules and create a session using your AWS credentials:
+
+```python
+import boto3
+
+session = boto3.Session(
+    aws_access_key_id='YOUR_ACCESS_KEY',
+    aws_secret_access_key='YOUR_SECRET_KEY',
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target="_blank">Click here for more details</a></b>